numpy/core/tests/test_datetime.py

```diff
@@ -2446,6 +2446,15 @@ def test_limit_str_roundtrip(self, time_unit, sign):
         limit_via_str = np.datetime64(str(limit), time_unit)
         assert limit_via_str == limit
 
+    def test_now_resolution_ns(self):
+        # test for gh-13372
+        stamp = np.datetime64(datetime.datetime.utcnow())
+        newer_stamp = np.datetime64("now", "ns")
+        # before supporting ns resolution, the
+        # second-truncated newer_stamp would
+        # fail this test
+        assert_array_less(stamp, newer_stamp)
```

Comment:
This regression test always failed on `main` for me locally, though it does occur to me that if you happen to run the test at x.99999 time units the clock might switch over and pass the test accidentally. I suspect the likelihood of that happening is on lottery ticket level probability, but haven't thought too hard about it.

There's probably a more obvious hook I could use to check--the repr not having all zeros at the end for example (although, you could accidentally have that pass as well if you just happen to hit 00000 ns or whatever).